src/library/features/crates/cratestoragehelpers.h

```diff
@@ -0,0 +1,239 @@
+#ifndef MIXXX_CRATESTORAGEHELPERS_H
+#define MIXXX_CRATESTORAGEHELPERS_H
+
+#include "library/crate/cratesummary.h"
+#include "library/crate/crateschema.h"
+#include "library/dao/trackschema.h"
+#include "track/trackid.h"
+
+#include "util/db/fwdsqlqueryselectresult.h"
+
+
+const QString CRATETABLE_LOCKED = "locked";
+
+const QString CRATE_SUMMARY_VIEW = "crate_summary";
+
+const QString CRATESUMMARY_TRACK_COUNT = "track_count";
+const QString CRATESUMMARY_TRACK_DURATION = "track_duration";
+
+const QString kCrateTracksJoin = QString(
+        "LEFT JOIN %3 ON %3.%4=%1.%2").arg(
+                CRATE_TABLE,
+                CRATETABLE_ID,
+                CRATE_TRACKS_TABLE,
+                CRATETRACKSTABLE_CRATEID);
+
+const QString kLibraryTracksJoin = kCrateTracksJoin + QString(
+        " LEFT JOIN %3 ON %3.%4=%1.%2").arg(
+                CRATE_TRACKS_TABLE,
+                CRATETRACKSTABLE_TRACKID,
+                LIBRARY_TABLE,
+                LIBRARYTABLE_ID);
+
+const QString kCrateSummaryViewSelect = QString(
+        "SELECT %1.*,"
+            "COUNT(CASE %2.%4 WHEN 0 THEN 1 ELSE NULL END) AS %5,"
+            "SUM(CASE %2.%4 WHEN 0 THEN %2.%3 ELSE 0 END) AS %6 "
+            "FROM %1").arg(
+                CRATE_TABLE,
+                LIBRARY_TABLE,
+                LIBRARYTABLE_DURATION,
+                LIBRARYTABLE_MIXXXDELETED,
+                CRATESUMMARY_TRACK_COUNT,
+                CRATESUMMARY_TRACK_DURATION);
+
+const QString kCrateSummaryViewQuery = QString(
+            "CREATE TEMPORARY VIEW IF NOT EXISTS %1 AS %2 %3 GROUP BY %4.%5").arg(
+                    CRATE_SUMMARY_VIEW,
+                    kCrateSummaryViewSelect,
+                    kLibraryTracksJoin,
+                    CRATE_TABLE,
+                    CRATETABLE_ID);
```

Comment:
I think it would be easier to maintain if these SQL string constants were put back into cratestorage.cpp (inside the anonymous namespace). Keep the helper classes in the new files.